README.md

```diff
@@ -0,0 +1,45 @@
+# piControl
+
+piControl is a kernel module for interfacing RevPi hardware. It provides a common interface to all RevPi related IOs, which a user can consume via `/dev/piControl0`.
+
+## Usage
+
+## Build the module
+
+All following steps need to be executed on a RevPi device running the official image. It is also possible to compile the module on another system, but the instructions may vary.
+
+Install kernel headers:
+
+```
+sudo apt update
+sudo apt install raspberrypi-kernel-headers
+```
+
+Checkout repository and switch working directory:
+
+```
+git clone https://github.com/RevolutionPi/piControl.git
+cd piControl
+```
+
+Compile module sources for current kernel and architecture:
+
+```
+KDIR=/usr/src/linux-headers-$(uname -r)/ make
+```
+
+## Load the compiled module
+
+Before the module can be loaded, the current module needs to unloaded:
+
+```
+sudo rmmod piControl
+```
+
+> **NOTE**: If the command fails with an error like `rmmod: ERROR: Module piControl is in use`, applications which are using the module needs to be stopped. The command `sudo lsof /dev/piControl0` will show a list of applications, which are using the piControl device.
+
+Load the kernel compiled module:
+
+```
+sudo insmod piControl.ko
```

Comment:
Probably worth noting that this will only load the newly compiled module once. On next reboot, the pre-installed module in /lib/modules/.../extra/ will be loaded again. You may want to explain how to install the newly compiled module so that it will be used permanently.